src/notification.c

```diff
@@ -133,6 +133,29 @@ void notification_run_script(struct notification *n)
                         if (pid2) {
                                 exit(0);
                         } else {
+                                // Set environment variables
+                                gchar *n_id = g_strdup_printf("%i", n->id);
+                                gchar *n_progress = g_strdup_printf("%i", n->progress);
+                                gchar *n_timeout = g_strdup_printf("%li", n->timeout);
+                                gchar *n_timestamp = g_strdup_printf("%li", n->timestamp);
+                                setenv("DUNST_APP_NAME",  appname,      true);
+                                setenv("DUNST_SUMMARY",   summary,      true);
+                                setenv("DUNST_BODY",      body,         true);
+                                setenv("DUNST_ICON",      icon,         true);
+                                setenv("DUNST_URGENCY",   urgency,      true);
+                                setenv("DUNST_ID",        n_id,         true);
+                                setenv("DUNST_PROGRESS",  n_progress,   true);
+                                setenv("DUNST_CATEGORY",  n->category,  true);
+                                setenv("DUNST_STACK_TAG", n->stack_tag, true);
+                                setenv("DUNST_URLS",      n->urls,      true);
+                                setenv("DUNST_TIMEOUT",   n_timeout,    true);
+                                setenv("DUNST_TIMESTAMP", n_timestamp,  true);
+                                setenv("DUNST_STACK_TAG", n->stack_tag, true);
+
+                                g_free(n_id);
+                                g_free(n_progress);
+                                g_free(n_timeout);
+                                g_free(n_timestamp);
```

Comment:
Are the frees necessary here? We're exec'ing out so everything will be lost either way.

Reply:
Yeah, you're right. I'll delete those frees. I'll delete the if after that as well, since that isn't needed as well.

Reply:
Why is the logic so complicated anyways? You only have to fork once right?

Reply:
Double forking forces a reparent of the child process to init. This means dunst doesn't have to wait to reap the exit code of the script (since we don't care).

Reply:
> I'll delete the if after that as well, since that isn't needed as well

If you're talking about the `if (ret != 0) {` one, that's needed in case `exec` fails :)

Reply:
>Double forking forces a reparent of the child process to init. This means dunst doesn't have to wait to reap the exit code of the script (since we don't care).

Ah that expains it. I'll but a comment there for clarity.

> If you're talking about the if (ret != 0) { one, that's needed in case exec fails :)

 Yeah, but the if statement itself is useless, since we only get there if exec fails.